code/game/objects/items/weapons/material/chainsaw.dm

```diff
@@ -0,0 +1,92 @@
+obj/item/weapon/material/twohanded/chainsaw
+	base_name = "Chainsaw"
+	desc = "Vroom vroom."
+	base_icon = "chainsaw"
+	var/on = 0
+	var/max_fuel = 30
+	w_class = ITEMSIZE_LARGE
+	slot_flags = SLOT_BACK
+	unwielded_force_divisor = 0.25
+	force_divisor = 0.5
+	dulled_divisor = 0.6
+	sharp = 1
+	edge = 1
+	w_class = ITEMSIZE_LARGE
+	slot_flags = SLOT_BACK
+	force_wielded = 35
+
+obj/item/weapon/material/twohanded/chainsaw/New()
+	var/datum/reagents/R = new/datum/reagents(max_fuel)
+	reagents = R
+	R.my_atom = src
+	R.add_reagent("fuel", max_fuel)
+	..()
+
+obj/item/weapon/material/twohanded/chainsaw/proc/turnOn()
+	if(on) return
+
+	to_chat(usr, "You start pulling the string on \the [src].")
+
+	if(max_fuel <= 0)
+		if(do_after(usr, 15))
+			to_chat(usr, "\The [src] won't start!")
+		else
+			to_chat(usr, "You fumble with the string.")
+	else if(!wielded)
+		to_chat(usr, "You need both hands to start the chainsaw!")
+	else
+		if(do_after(usr, 15))
+			to_chat(usr, "\The [src] starts up with a loud noise!")
```

Comment:
Should probably be more than just to usr, no?

Reply:
Probably.